src/js/profile.js

```diff
@@ -87,10 +87,24 @@ $(function() {
                 .fail(xhr => complain(xhr.responseText))
         })
 
-        $("#input0").autocomplete({
-            source: friends,
-            appendTo: "#modal"
+        $("#input0").attr("data-list", friends );
+
+        new Awesomplete( ("#input0"), {
+            
+            filter: function(text, input) {
+                return Awesomplete.FILTER_CONTAINS(text, input.match(/[^,]*$/)[0]);
```

Comment:
I see what you mean about the self.email part. Could we not set the default being queried is the users email unless specified?
Everything else has worked fine afaik